vdr/didsubject/manager.go

```diff
@@ -522,3 +527,44 @@ func (r *Manager) Rollback(ctx context.Context) {
 		log.Logger().WithError(err).Error("failed to rollback DID documents")
 	}
 }
+
+func sortDIDs(list []did.DID, order []string) {
+	// Thank you, Copilot
+	sort.Slice(list, func(i, j int) bool {
+		iOrder := -1
+		jOrder := -1
+		for k, v := range order {
+			if v == list[i].Method {
+				iOrder = k
+			}
+			if v == list[j].Method {
+				jOrder = k
+			}
+		}
+		// If both are -1, they are not in the preferred order list, so sort by method for stable order
+		if iOrder == -1 && jOrder == -1 {
+			return list[i].Method < list[j].Method
+		}
+		return iOrder < jOrder
+	})
+}
+
+func sortDIDDocuments(list []did.Document, order []string) {
+	listOfDIDs := make([]did.DID, len(list))
+	for i, doc := range list {
+		listOfDIDs[i] = doc.ID
+	}
+	sortDIDs(listOfDIDs, order)
+	// order list according to listOfDIDs
+	orderedList := make([]did.Document, len(list))
+	for i, id := range listOfDIDs {
+	inner:
+		for _, doc := range list {
+			if doc.ID == id {
+				orderedList[i] = doc
+				break inner
+			}
+		}
+	}
```

Comment:
This fails if `listOfDIDs` contains duplicates. Add a godoc describing this incase we re-use the method later in a scenario where this is possible

Reply:
would be weird, but fixed it just in kees.